packages/jest-config/src/readConfigFileAndSetRootDir.ts

```diff
@@ -143,8 +150,8 @@ async function registerTsLoader(loader: TsLoaderModule): Promise<TsLoader> {
         moduleTypes: {
           '**': 'cjs',
         },
-        transpileOnly:
-          process.env.JEST_CONFIG_TRANSPILE_ONLY?.toLowerCase() === 'true',
+        swc: extraTSLoaderOptions?.swc,
+        transpileOnly: extraTSLoaderOptions?.transpileOnly,
```

Comment:
Instead of setting these two, can we just spread the config passed?

i.e.

```ts
tsLoader.register({
  compilerOptions: {
    module: 'CommonJS',
  },
  moduleTypes: {
    '**': 'cjs',
  },
  ...extraTSLoaderOptions,
});
```

Reply:
I thought the same but not sure if it's a good idea to expose all options or not, done [4223489](https://github.com/jestjs/jest/pull/15234/commits/422348953d0cde3f0d751002a43fc3871345cd5c)

Reply:
if people misconfigure, that's on them. I think exposing the API is better than having to expose more in the future